execution/scan/matrix_selector.go

```diff
@@ -366,10 +368,20 @@ loop:
 		case chunkenc.ValNone:
 			break loop
 		case chunkenc.ValHistogram, chunkenc.ValFloatHistogram:
+			t, h := buf.AtHistogram()
```

Comment:
Hmm, we call `AtFloatHistogram` bellow, so I wonder why we need to call `AtHistogram` here. We also seem to now append to the output before we check for staleness, so we might end up picking up stale samples. Is this the correct behavior?

Reply:
You are correct, this was an artifact of rebasing. I somehow missed this part. Thank you for a diligent review :bow: